### ISSUE TITLE
Error using pre-trained model #Enhancement

### ISSUE THREAD

**Issue by FlorianeSchreiber** [remicres/sr4rs#37]
Hello @remicres,

I would like to use the pre-trained model, but I have an error. I followed the "How to use?" and installed all dependancies. 
I downloaded from Sentinel-Hub the 4 bands (4, 3, 2 and 8  in TIFF Format) from my image, and use otbConcatenate to have an image with these 4 bands. 
I launched this command line : 

```
python3 sr4rs/code/sr.py \
--savedmodel sr4rs_sentinel2_bands4328_france2020_savedmodel \
--input otbConcatenateImages.tif \
--output SalonProvence_AugmentedRes.tif
```
and I have this error, and don't know how to fix it ...  : 

```
Traceback (most recent call last):
  File "sr4rs/code/sr.py", line 61, in <module>
    infer.SetParameterStringList("source1.il", [params.input])
AttributeError: 'NoneType' object has no attribute 'SetParameterStringList'
```
<img width="659" alt="Capture d’écran 2021-10-26 à 16 07 21" src="https://user-images.githubusercontent.com/80046066/138895932-0c851199-0c99-46ca-a5e6-8c7aeeac9695.png">

Thank you for your answer,

F.Schreiber

**Comment by remicres** [comment]
Hi @FlorianeSchreiber,

This message pops when the *TensorflowModelServe* application does not exist in otb. 
I understand that you didn't use the docker image, and install everything from source ? In this case, you have to stick with the compilation settings provided in the dockerfile.

**Comment by FlorianeSchreiber** [comment]
Hi,

Ok, that is the reason why !
Yes, I didn't use the docker image, because I am not able to install Nvidia/CUDA drivers (I am on MacOs...).
Ok, I will do that ! Thank you Rémi !

**Comment by remicres** [comment]
For CPU-only, you can use the **cpu-xxx** tagged docker images.

Processing SR4RS on CPU is slow though... it will run faster on GPU !

**Comment by FlorianeSchreiber** [comment]
Ok, I used `docker run -it mdl4eo/otbtf2.5:cpu bash` finally. I copied all folders and files into the otbuser bash, the code starts, and after that, I have a new error 😞 : 

> python sr4rs/code/sr.py --savedmodel sr4rs_sentinel2_bands4328_france2020_savedmodel --input otbConcatenateImages_bis.tif --output test.tif

> 2021-11-04 08:16:58 (INFO) ReadImageInfo: 
> Image general information:
> 	Number of bands : 4
> 	Data type : float
> 	No data flags : Not found
> 	Start index :  [0,0]
> 	Size :  [2500,1163]
> 	Origin :  [550195,5.41389e+06]
> 	Spacing :  [11.0069,-11.0088]
> 	Estimated ground spacing (in meters): [7.96303,7.96005]
> 
> Image acquisition information:
> 	Sensor : 
> 	Image identification number: 
> 	Image projection : PROJCS["WGS 84 / Pseudo-Mercator",
>     GEOGCS["WGS 84",
>         DATUM["WGS_1984",
>             SPHEROID["WGS 84",6378137,298.257223563,
>                 AUTHORITY["EPSG","7030"]],
>             AUTHORITY["EPSG","6326"]],
>         PRIMEM["Greenwich",0,
>             AUTHORITY["EPSG","8901"]],
>         UNIT["degree",0.0174532925199433,
>             AUTHORITY["EPSG","9122"]],
>         AUTHORITY["EPSG","4326"]],
>     PROJECTION["Mercator_1SP"],
>     PARAMETER["central_meridian",0],
>     PARAMETER["scale_factor",1],
>     PARAMETER["false_easting",0],
>     PARAMETER["false_northing",0],
>     UNIT["metre",1,
>         AUTHORITY["EPSG","9001"]],
>     AXIS["Easting",EAST],
>     AXIS["Northing",NORTH],
>     EXTENSION["PROJ4","+proj=merc +a=6378137 +b=6378137 +lat_ts=0 +lon_0=0 +x_0=0 +y_0=0 +k=1 +units=m +nadgrids=@null +wktext +no_defs"],
>     AUTHORITY["EPSG","3857"]]
> 
> Image default RGB composition:
> 	[R, G, B] = [0,1,2]
> 
> Ground control points information:
> 	Number of GCPs = 0
> 	GCPs projection = 
> 
> Illegal instruction

Is a problem with the image ? With the projection ? 

Thank you :-)

**Comment by remicres** [comment]
Maybe some optimization flags in `mdl4eo/otbtf2.5:cpu` aren't supported by your CPU.

Can you give a try with `mdl4eo/otbtf2.5:cpu-basic` ? In this docker image, TensorFlow does not use modern CPU optimization flags (no AVX, etc.).

**Comment by FlorianeSchreiber** [comment]
It is better, but ... finally it crashed ! 

`2021-11-04 09:16:03 (INFO) TensorflowModelServe: Default RAM limit for OTB is 256 MB
2021-11-04 09:16:03 (INFO) TensorflowModelServe: GDAL maximum cache size is 99 MB
2021-11-04 09:16:03 (INFO) TensorflowModelServe: OTB will use at most 4 threads
2021-11-04 09:16:03.623346: I tensorflow/cc/saved_model/reader.cc:38] Reading SavedModel from: sr4rs_sentinel2_bands4328_france2020_savedmodel
2021-11-04 09:16:04.223736: I tensorflow/cc/saved_model/reader.cc:90] Reading meta graph with tags { serve }
2021-11-04 09:16:04.223826: I tensorflow/cc/saved_model/reader.cc:132] Reading SavedModel debug info (if present) from: sr4rs_sentinel2_bands4328_france2020_savedmodel
2021-11-04 09:16:04.224730: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE3 SSE4.1 SSE4.2 AVX AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-11-04 09:16:06.246332: I tensorflow/cc/saved_model/loader.cc:206] Restoring SavedModel bundle.
2021-11-04 09:16:09.524312: I tensorflow/core/platform/profile_utils/cpu_utils.cc:114] CPU Frequency: 2000000000 Hz
2021-11-04 09:17:19.123524: I tensorflow/core/profiler/lib/profiler_session.cc:126] Profiler session initializing.
2021-11-04 09:17:19.132221: I tensorflow/core/profiler/lib/profiler_session.cc:141] Profiler session started.
Killed`

**Comment by remicres** [comment]
Ah the hardware.

Try to reduce the memory footprint of the processing, changing small values to `--ts` (e.g. 64) and `--pads` (try 16).

**Comment by FlorianeSchreiber** [comment]
Pad value could be only 32, 64, 128, 256 or 512. But I tried with 32, and for the ts value, I tried several value (64 and less). I reduced a lot the size of the size image. And it continues to kill... !! 
Ha ha, the computer is modern, but with another OS, it would be better, indeed 😆.
In any way, thank you for your help !!

**Comment by remicres** [comment]
out of curiosity, what is your system and OS?

**Comment by FlorianeSchreiber** [comment]
I was on MacOS BigSur yesterday and Monterey since today. I tried both !

**Comment by remicres** [comment]
My first guess is that it could be some compilation flags or settings in bazel, that makes the docker image incompatible with MacOS.

But it's clearly hypothetical. We don't have very much feedback from mac users :disappointed:

**Comment by FlorianeSchreiber** [comment]
Finally the problem was in my docker parameters. It was limited, and when I modified these parameters, it works fine.
The place where the parameters that I modified is : Parameters => Resourced => Advanced.
(I was limited to 2 GB RAM and 4 CPU)

<img width="1139" alt="Capture d’écran 2021-12-01 à 09 06 26" src="https://user-images.githubusercontent.com/80046066/144195603-18a63d54-79ff-4bcf-86c9-654ced11e7da.png">
. 

Thank you @remicres ;-)
